File: 1827-minimum-operations-to-make-the-array-increasing/1827-minimum-operations-to-make-the-array-increasing.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
       int res = 0, last = 0;
    for (auto n : nums) {
        res += max(0, last - n + 1);
        last = max(n, last + 1);
    }
    return res;  
    }
};

File: 1827-minimum-operations-to-make-the-array-increasing/1827-minimum-operations-to-make-the-array-increasing.java
class Solution {
    public int minOperations(int[] nums) {
         int res = 0, last = 0;
    for (int n : nums) {
        res += Math.max(0, last - n + 1);
        last = Math.max(n, last + 1);
    }
    return res;
    }
}